package/network/config/ltq-vdsl-app/files/10_atm_ptm.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/sh
+
+[ "$DSL_NOTIFICATION_TYPE" = "DSL_STATUS" ] || exit 0
+
+if [ "$DSL_TC_LAYER_STATUS" = "ATM" ]; then
+
+	if grep -q "ltq_ptm_vr9" /proc/modules ; then
+		# TODO: make ltq_atm_vr9/ltq_ptm_vr9 "hot-swappable"
+		# ifconfig ptm0 down &>/dev/null
+		# rmmod "ltq_ptm_vr9"
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer ATM while PTM driver is loaded. Rebooting System"
+		reboot
+		exit
+	fi
+
+	if ! grep -q "ltq_atm_vr9" /proc/modules ; then
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer ATM"
+
+		modprobe ltq_atm_vr9
+		esi $(printf '%012X' $((1+0x$(tr -d : </sys/class/net/eth0/address))))
+		
+		grep -q "br2684" /proc/modules || modprobe br2684
+		/etc/init.d/br2684ctl start &>/dev/null
+		ip link set dev nas0 up
+	fi
+
+elif [ "$DSL_TC_LAYER_STATUS" = "EFM" ]; then
+	if grep -q "br2684" /proc/modules ; then
+		ifconfig nas0 down &>/dev/null
+		/etc/init.d/br2684ctl stop &>/dev/null
+		rmmod "br2684"
+	fi
+
+	if grep -q "ltq_atm_vr9" /proc/modules ; then
+		# TODO: make ltq_atm_vr9/ltq_ptm_vr9 "hot-swappable"
+		# rmmod "ltq_atm_vr9"
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer EFM/PTM while ATM driver is loaded. Rebooting System"
+		reboot
+		exit
+	fi
+
+	if ! grep -q "ltq_ptm_vr9" /proc/modules ; then
+		logger -p daemon.notice -t "dsl-notify" "Switching to TC-Layer EFM/PTM"
+
+		modprobe ltq_ptm_vr9
+		ip link set dev ptm0 down
+		ip link set dev ptm0 address $(printf '%012X' $((1+0x$(tr -d : </sys/class/net/eth0/address))))
```

Comment:
Depending on the configuration, this would set a different mac address than the one configured with ```ucidef_set_interface_macaddr()```. Isn't the mac address set by netifd if the new netdev is registered?